lib/chef/provisioning/aws_driver/aws_resource.rb

```diff
@@ -118,6 +122,12 @@ def self.get_aws_object(value, resource: nil, run_context: nil, driver: nil, man
       resource = value
     else
       resource = new(value, run_context)
+
+      # populate AWS ID for resources like AwsRoute53HostedZone that don't have lazy-auto assignment.
+      if resource.public_send(resource.class.aws_id_attribute).nil?
```

Comment:
I don't think this is necessary - we overwrote `self.attribute` to set the `aws_id_attribute` to whatever attribute name is marked.